presto-native-execution/presto_cpp/main/operators/LocalPersistentShuffle.cpp

```diff
@@ -127,38 +142,35 @@ void LocalPersistentShuffleWriter::noMoreData(bool success) {
       storePartitionBlock(i);
     }
   }
-  // Write "ready to read" file.
-  auto readyToRead = fileSystem_->openFileForWrite(
-      fmt::format("{}/{}", rootPath_, kReadyForReadFilename));
-  readyToRead->close();
 }
 
 LocalPersistentShuffleReader::LocalPersistentShuffleReader(
     const std::string& rootPath,
+    const std::string& queryId,
+    std::vector<std::string> partitionIds,
     const int32_t partition,
     velox::memory::MemoryPool* FOLLY_NONNULL pool)
-    : rootPath_(std::move(rootPath)), partition_(partition), pool_(pool) {
+    : rootPath_(rootPath),
+      queryId_(queryId),
+      partitionIds_(std::move(partitionIds)),
+      partition_(partition),
+      pool_(pool) {
   fileSystem_ = velox::filesystems::getFileSystem(rootPath_, nullptr);
 }
 
 bool LocalPersistentShuffleReader::hasNext() {
-  while (!readyForRead()) {
-    // This sleep is only for testing purposes.
-    // For the test cases in which the shuffle reader tasks run before shuffle
-    // writer tasks finish.
-    sleep(1);
+  if (readPartitionFiles_.empty()) {
```

Comment:
What happens if shuffle data is empty?

Reply:
if the shuffle data it empty, the hasNext will return false indicating the shuffle read is finished.  In spark, it's guaranteed to only run shuffle read job after the shuffle write jobs are done (shuffle data exists).